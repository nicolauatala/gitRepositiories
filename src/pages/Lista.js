import React, { useState, useEffect } from 'react';
import { SafeAreaView, View, Text, TextInput, StyleSheet, TouchableOpacity } from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';
import CustomListview from '../components/CustomListview';

import api from "../services/api";
import EmptyList from '../components/EmptyList';
export default class Lista extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            searchInput: '',
            repositories: []
        };
    }

    static navigationOptions = {
        title: 'GitIssues',
    };

    async componentDidMount() {
        // await AsyncStorage.clear();
        const repos = await AsyncStorage.getItem('@Store:repositories');
        if (repos) {
            this.setState({repositories: JSON.parse(repos)})
        }
    }

    render() {

        const saveRepository = async (repository) => {
            
            try {
                AsyncStorage.getItem('@Store:repositories')
                    .then((repositories) => {
                        const newRepositories = repositories ? JSON.parse(repositories) : [];
                        newRepositories.push(repository);
                    AsyncStorage.setItem('@Store:repositories', JSON.stringify(newRepositories));
                });    
            } catch (error) {
                console.log('error', error);
            }
        }

        const searchRepository = async () => {
            const response = await api.get(`/repos/${this.state.searchInput}`);
            if (response.data) {
                const { id, name, owner } = response.data;
                const newelement = {
                    id: id,
                    title: name,
                    description: owner.login,
                    image_url: owner.avatar_url
                }
                this.setState(prevState => ({
                    repositories: [...prevState.repositories, newelement]
                }));
                saveRepository(newelement);
                this.setState({searchInput : ''});
            }
        }

        return (
            <SafeAreaView style={styles.container}>
                <View style={styles.searchContainer}>
                    <TextInput 
                        autoCapitalize="none"
                        autoCorrect={false}
                        placeholder="Digite seu usuÃ¡rio no Github"
                        style={styles.search}
                        value={this.state.searchInput}
                        onChangeText={(text) => this.setState({searchInput : text})}/>
                    <TouchableOpacity
                        onPress={searchRepository}
                        style={styles.searchButton}>
                        <Text style={styles.searchButtonText}>+</Text>
                    </TouchableOpacity>
                </View>
                
                {
                    this.state.repositories.length === 0 ?
                        <EmptyList />
                    :  <CustomListview itemList={this.state.repositories} />
                }
            </SafeAreaView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#ededed',
    },

    empty: {
        fontSize: 22
    },  

    searchContainer: {
        flexDirection: 'row',
        alignContent: 'center',
        justifyContent: 'center',
        borderBottomColor: '#CCC',
        marginHorizontal: 16,
        borderBottomWidth: 1,
        marginBottom: 10,
        paddingBottom: 8
    },

    search: {
        flex: 5,
        height: 46,
        marginBottom: 10,
        alignSelf: 'stretch',
        backgroundColor: '#FFF',
        borderWidth: 1,
        borderColor: '#ddd',
        borderRadius: 6,
        marginTop: 20,
        paddingHorizontal: 15,
        
    },

    searchButton: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
    },

    searchButtonText: {
        fontWeight: 'bold',
        fontSize: 28,
    },
    
    repositoryItem: {
        backgroundColor: '#FFF',
        height: 74,
        borderRadius: 8,
        marginBottom: 20,
        marginHorizontal: 20
    }

});