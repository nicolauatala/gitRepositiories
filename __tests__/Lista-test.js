import React from 'react';
import Lista from '../src/pages/Lista';
import api from "../src/services/api";

import renderer from 'react-test-renderer';

beforeAll(() => { 
    jest.mock('@react-native-community/async-storage');
});

it('renders correctly', () => {
    const tree = renderer.create(<Lista />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('works with async/await', async () => {
    const response = await api.get(`/repos/nicolauatala/punkbeer`);
    const { owner, name } = response.data;
    expect(owner.login).toEqual('nicolauatala');
    expect(name).toEqual('punkBeer');
});