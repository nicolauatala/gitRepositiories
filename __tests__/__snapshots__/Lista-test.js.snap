// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ededed",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "borderBottomColor": "#CCC",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "marginHorizontal": 16,
        "paddingBottom": 8,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Digite seu usuÃ¡rio no Github"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#FFF",
          "borderColor": "#ddd",
          "borderRadius": 6,
          "borderWidth": 1,
          "flex": 5,
          "height": 46,
          "marginBottom": 10,
          "marginTop": 20,
          "paddingHorizontal": 15,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 28,
            "fontWeight": "bold",
          }
        }
      >
        +
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#666",
          "fontSize": 64,
          "fontWeight": "bold",
        }
      }
    >
      +
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#999",
        }
      }
    >
      No repository added :/
    </Text>
  </View>
</RCTSafeAreaView>
`;
